docs/WalletGuide.md

```diff
@@ -699,3 +707,224 @@ suspend fun main() {
+
+### Create recovery class
+First, we need to define recovery server that will be used. For simplicity, there is a key assigned to each server, so they can be referenced later. 
+
+```kotlin
+val first = RecoveryServerKey("first")
+val second = RecoveryServerKey("second")
+val firstServer = RecoveryServer("recovery.example.com", "auth.example.com", "example.com")
+val secondServer = RecoveryServer("recovery2.example.com", "auth2.example.com", "example.com")
+val servers = mapOf(first to firstServer, second to secondServer)
+val recovery = wallet.recovery(servers)
+```
+
+> You can get the full code [here](../examples/documentation/src/example-recovery-01.kt).
+
+<!--- INCLUDE .*recovery.*
+val first = RecoveryServerKey("first")
+val second = RecoveryServerKey("second")
+val firstServer = RecoveryServer("recovery.example.com", "auth.example.com", "example.com")
+val secondServer = RecoveryServer("recovery2.example.com", "auth2.example.com", "example.com")
+val servers = mapOf(first to firstServer, second to secondServer)
+val recovery = wallet.recovery(servers)
+val sponsor = SigningKeyPair.fromSecret("<example key>")
+-->
+
+### Create recoverable account
+
+<!--- INCLUDE 
+suspend fun createAccount() {
+-->
+
+<!--- SUFFIX 
+}
+-->
+
+First, let's create an account key and a device key that will be attached to this account
+```kotlin
+  val account = wallet.stellar().account().createKeyPair()
+  val deviceKey = wallet.stellar().account().createKeyPair()
+```
+
+Next, we need to define SEP-30 identities. In this example we are going to use the same list of identities on both recovery servers. 
+```kotlin
+  val recoveryKey = wallet.stellar().account().createKeyPair()
+
+  val identities =
+    listOf(
+      RecoveryAccountIdentity(
+        RecoveryRole.OWNER,
+        listOf(RecoveryAccountAuthMethod(RecoveryType.STELLAR_ADDRESS, recoveryKey.address))
+      )
+    )
+```
+Here, stellar key will be used as a recovery method. Other recovery servers may support email or phone as a recovery methods
+
+You can read more about SEP-30 identities [here](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0030.md#common-request-fields)
+
+Next, let's create a recoverable account:
+```kotlin
+  val recoverableWallet =
+    recovery.createRecoverableWallet(
+      RecoverableWalletConfig(
+        account,
+        deviceKey,
+        AccountThreshold(10, 10, 10),
+        mapOf(first to identities, second to identities),
+        SignerWeight(10, 5),
+        sponsor
+      )
+    )
+```
+With given parameters, this function will:
+1. Create `account` on the Stellar network (if it's not created yet).
+2. `sponsor` will be used to sponsor all operations. You may not sponsor creating recoverable wallet, but in that case `account` must already be created.
+3. `deviceKey` will be used as a primary account key. Please note that master key belonging to `account` will be locked. `deviceKey` should be used as a primary signer instead.
```

Comment:
Isn't this pretty opinionated? Couldn't an account use its master key as the device key instead of generating a new one? I understand this is how vibrant did it but I never understood why.

Reply:
What happens if I pass the account's master secret key in as the device key?

Reply:
I decided to leave it like this for now. It's pretty common approach, and the good thing about it is that it's well paired with a method that creates transaction for signer recovery.
I was thinking about making it optional, but it will complicate implementation. We may add it in the future vesions

Reply:
> What happens if I pass the account's master secret key in as the device key?

It will throw an error, but probably not very informative. I will look into it.